### PR TITLE
Update OpenAPI SecurityDefinition to use Authorization header name, following HTTP standard

### PR DESCRIPTION
**Pull request by saurav-malani**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Update OpenAPI SecurityDefinition to use Authorization header name, following HTTP standard

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This is done in reference to https://github.com/kubernetes-client/python/issues/128.
          So, as per the python client, issue-128 kubernetes python client uses authorization header key with small "a", whereas most of the python client libraries use capital "A" Authorization header key. So, here we tracked the source of this small "a" authorization and made changes in pkg/kubeapiserver/authenticator/config.go, staging/src/k8s.io/apiserver/pkg/authentication/authenticatorfactory/delegating.go, pkg/kubectl/apply/strategy/test_swagger.json, and then hack/update-openapi-spec.sh is run, which regenerated /api/openapi-spec/swagger.json with the corresponding changes, incorporating capital "A" Authorization in place of small "a" authorization header key.

**Does this PR introduce a user-facing change?**:
yes
```release-note
Update OpenAPI SecurityDefinition to use Authorization header name, following HTTP standard.
```

### CODE REVIEW

**Comment by roycaihw** [comment]
LGTM from RFCs and OpenAPI perspective, as https://github.com/kubernetes-client/python/issues/128#issuecomment-281459813 suggested. Since HTTP header field names are case-insensitive this change should be backwards compatible (the change is mostly documentation anyway).

/sig auth

Please add a release note to the first comment as the template suggested. Also the current PR title seems to be more related to the python client. I'd suggest changing to something like "Update OpenAPI SecurityDefinition to use `Authorization` header name, following HTTP standard" for this repo.

File: pkg/kubectl/apply/strategy/test_swagger.json [hunk 1]
```diff
@@ -325,16 +325,16 @@
...
+   }
   },
   "security": [
     {
       "BearerToken": []
     }
   ]
-}
\ No newline at end of file
+
```
Comment (roycaihw) [inline pkg/kubectl/apply/strategy/test_swagger.json]: The apply test failed because the `}` was missed

**Comment by mourya007** [comment]
I guess you can now squash commits.
/test pull-kubernetes-godeps

**Comment by mourya007** [comment]
I think you can squash your commits now.

**Comment by mourya007** [comment]
cc @liggitt

**Comment by mourya007** [comment]
I think you can now squash your commits.
/test pull-kubernetes-godeps

**Comment by liggitt** [comment]
/hold

needs to be squashed

**Comment by liggitt** [comment]
closing due to inactivity. feel free to squash and reopen

/close

**Pull request closed without merging**